src/components/new-request/business-fetch.vue

```diff
@@ -40,45 +38,54 @@ enum States {
  */
 @Component({})
 export default class BusinessFetch extends Vue {
-  // Refs
-  $refs!: {
-    searchField: FormType
-  }
-
-  // enum for template
+  // Enum for template
   readonly States = States
 
+  // Store action
+  @Action fetchCorpNum!: (corpNum: string) => Promise<BusinessFetchIF>
+
   /** V-model for search field. */
   searchField = ''
 
+  /** For custom error messages. */
+  errorMessages = [] as Array<string>
+
   /** State of this component. */
   state = States.INITIAL
 
-  /** Validation rules. */
-  rules = [
-    v => !!v || 'Required field',
-    v => (!v || v.length === 9) || 'Please enter a valid registration number'
-  ]
+  /** Validates the search field, sets any error messages, and returns validity. */
+  private validate (): boolean {
+    if (!this.searchField) {
+      this.errorMessages = ['Required field']
+    } else if (!/^(A|BC|C|CP|FM)\d{7}$/.test(this.searchField)) {
+      this.errorMessages = ['Please enter a valid registration number']
+    } else {
+      this.errorMessages = []
+    }
+    return (this.errorMessages.length === 0)
+  }
```

Comment:
I couldn't get rules-based validation to not validate while the user was typing (even when I wrapped the v-text-field in a v-form with lazy-validation) so I did this instead.